### DISCUSSION TITLE
Batching Transactions

### DISCUSSION THREAD

**Discussion by groogiam** [OData/AspNetCoreOData#490] [Q&A]
Is there any documentation or guidance on how to implement transactions across and odata batch? Essentially I'm looking to have all operations submitted in a batch share an EF Core IDbContextTransaction or System.Transactions.TransactionScope.

**Comment by bbatchelder** [comment]
Did you get anywhere with this....I have the same question :-)

**Reply by groogiam** [reply]
@bbatchelder I have not really had much time to work on this on this since I posted the question. I suspect that the answer is implementing a custom batch handler instead of using `DefaultODataBatchHandler` and adding the transaction logic in there somewhere.  It would be nice if the OData team could provide some documentation of this as seems like a pretty common use case. I'll be sure to update this issue when I finally get around to implementing this if no one else has provided any additional details. Sorry I could not be of more help.
